### PR TITLE
8321542: C2: Missing ChaCha20 stub for x86_32 leads to crashes

### PR DESCRIPTION
**Pull request by jnimeh**
This fix corrects an oversight in the ChaCha20 intrinsics delivered by JDK-8247645.  An ifdef guard is now part of the x86 ChaCha20 intrinsics code which disables them by default on 32-bit platforms, as this architecture was not part of the feature delivery.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8321542](https://bugs.openjdk.org/browse/JDK-8321542): C2: Missing ChaCha20 stub for x86_32 leads to crashes (**Bug** - P3)

### Reviewers
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**) ⚠️ Review applies to [fe0d4461](https://git.openjdk.org/jdk/pull/17072/files/fe0d4461096c2e239acd58077bd4bfe61f90ec7f)
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**) ⚠️ Review applies to [fe0d4461](https://git.openjdk.org/jdk/pull/17072/files/fe0d4461096c2e239acd58077bd4bfe61f90ec7f)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17072/head:pull/17072` \
`$ git checkout pull/17072`

Update a local copy of the PR: \
`$ git checkout pull/17072` \
`$ git pull https://git.openjdk.org/jdk.git pull/17072/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17072`

View PR using the GUI difftool: \
`$ git pr show -t 17072`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17072.diff">https://git.openjdk.org/jdk/pull/17072.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17072#issuecomment-1851144891)

### CODE REVIEW

**Comment by jnimeh** [comment]
The bug on 32-bit without this fix is easily reproducible by running any of the ChaCha20 microbenchmarks, or via the AEADBufferTest microbenchmark.  Likewise, once the fix is applied these benchmarks no longer crash the JVM.

File: src/hotspot/cpu/x86/vm_version_x86.cpp [hunk 1]
```diff
@@ -1145,6 +1146,13 @@ void VM_Version::get_processor_features() {
       }
       FLAG_SET_DEFAULT(UseChaCha20Intrinsics, false);
   }
+#else
+  // No support currently for ChaCha20 intrinsics on 32-bit platforms
+  if (UseChaCha20Intrinsics) {
+      warning("Support for ChaCha20 intrinsics not available on this CPU.");
```
Comment (chhagedorn) [inline src/hotspot/cpu/x86/vm_version_x86.cpp:1152]: Maybe we can adapt the message to follow the convention of the other warning messages for intrinsics like AES
https://github.com/openjdk/jdk/blob/2611a49ea13ee7a07f22692c3a4b32856ec5898f/src/hotspot/cpu/x86/vm_version_x86.cpp#L1062-L1065
or CRC32C
https://github.com/openjdk/jdk/blob/2611a49ea13ee7a07f22692c3a4b32856ec5898f/src/hotspot/cpu/x86/vm_version_x86.cpp#L1116-L1118

```suggestion
      warning("ChaCha20 intrinsics are not available on this CPU.");
```

Though I see that not all existing warning messages are consistent.

Anyway, the bailout looks good to me.

**Review by chhagedorn — APPROVED** [review]

File: src/hotspot/cpu/x86/vm_version_x86.cpp (hunk 1 above)
Comment (shipilev) [reply src/hotspot/cpu/x86/vm_version_x86.cpp:1152]: +1 on changing the message to "ChaCha20 intrinsics are not available on this CPU."

**Review by shipilev — APPROVED** [review]
Agreed with message suggestion, otherwise good.

File: src/hotspot/cpu/x86/vm_version_x86.cpp (hunk 1 above)
Comment (jnimeh) [reply src/hotspot/cpu/x86/vm_version_x86.cpp:1152]: Makes sense to me.  Will fix.

**Pull request closed without merging**